src/index.js

```diff
@@ -91,6 +100,7 @@ export default function fetch(url, opts) {
 
 		req.once('response', res => {
 			clearTimeout(reqTimeout);
+			removeSignalListeners();
```

Comment:
`removeSignalListeners()` is called on response and in `finallize()`. This should remove all signal events for `Request` object and `fetch` itself.

Reply:
Hm, why are we removing the event listeners on response? We want the abort controller to work even after the initial response has been received… or am I missing something?

Reply:
Since the current Request has been resolved, we must remove it from signal's event listeners so it can be garbage collected. Unless I missed something, Response cannot be aborted. I did, however, forgot to propagate signal when redirecting.